dotnet/src/Connectors/Connectors.AzureOpenAI/Core/ClientCore.cs

```diff
@@ -1571,4 +1292,95 @@ private void LogUsage(CompletionsUsage usage)
             await functionCallCallback(context).ConfigureAwait(false);
         }
     }
+
+    private ToolCallingConfig GetToolCallingConfiguration(Kernel? kernel, AzureOpenAIPromptExecutionSettings executionSettings, int requestIndex)
+    {
+        if (executionSettings.ToolCallBehavior is null)
+        {
+            return new ToolCallingConfig(Tools: [s_nonInvocableFunctionTool], Choice: ChatToolChoice.None, AutoInvoke: false);
+        }
+
+        if (requestIndex >= executionSettings.ToolCallBehavior.MaximumUseAttempts)
+        {
+            // Don't add any tools as we've reached the maximum attempts limit.
+            if (this.Logger.IsEnabled(LogLevel.Debug))
+            {
+                this.Logger.LogDebug("Maximum use ({MaximumUse}) reached; removing the tool.", executionSettings.ToolCallBehavior!.MaximumUseAttempts);
+            }
+
+            return new ToolCallingConfig(Tools: [s_nonInvocableFunctionTool], Choice: ChatToolChoice.None, AutoInvoke: false);
+        }
+
+        var (tools, choice) = executionSettings.ToolCallBehavior.ConfigureOptions(kernel);
+
+        bool autoInvoke = kernel is not null &&
+            executionSettings.ToolCallBehavior.MaximumAutoInvokeAttempts > 0 &&
+            s_inflightAutoInvokes.Value < MaxInflightAutoInvokes;
+
+        // Disable auto invocation if we've exceeded the allowed limit.
+        if (requestIndex >= executionSettings.ToolCallBehavior.MaximumAutoInvokeAttempts)
+        {
+            autoInvoke = false;
+            if (this.Logger.IsEnabled(LogLevel.Debug))
+            {
+                this.Logger.LogDebug("Maximum auto-invoke ({MaximumAutoInvoke}) reached.", executionSettings.ToolCallBehavior!.MaximumAutoInvokeAttempts);
+            }
+        }
+
+        return new ToolCallingConfig(
+            Tools: tools ?? [s_nonInvocableFunctionTool],
+            Choice: choice ?? ChatToolChoice.None,
+            AutoInvoke: autoInvoke);
+    }
+
+    private static ChatResponseFormat? GetResponseFormat(AzureOpenAIPromptExecutionSettings executionSettings)
```

Comment:
nit.

Might consider using switch expressions to reduce the verbosity.

```suggestion
    private static ChatResponseFormat? GetResponseFormat(AzureOpenAIPromptExecutionSettings executionSettings)
```

Reply:
I might do it some other time, there are more important things to finish.